spec/requests/users_controller_spec.rb

```diff
@@ -1,51 +1,2402 @@
+
+          expect(JSON.parse(response.body)['active']).to be_falsy
+        end
+      end
+    end
+
+    context "creating as staged" do
+      it "won't create the user as staged" do
+        post "/u.json", params: post_user_params.merge(staged: true)
+
+        new_user = User.where(username: post_user_params[:username]).first
+        expect(new_user.staged?).to eq(false)
+      end
+
+      context "with a regular api key" do
+        let(:user) { Fabricate(:user) }
+        let(:api_key) { Fabricate(:api_key, user: user) }
+
+        it "won't create the user as staged with a regular key" do
+          post "/u.json", params: post_user_params.merge(staged: true, api_key: api_key.key)
+
+          new_user = User.where(username: post_user_params[:username]).first
+          expect(new_user.staged?).to eq(false)
+        end
+      end
+
+      context "with an admin api key" do
+        let(:user) { Fabricate(:admin) }
+        let(:api_key) { Fabricate(:api_key, user: user) }
+
+        it "creates the user as staged with a regular key" do
+          post "/u.json", params: post_user_params.merge(staged: true, api_key: api_key.key)
+
+          new_user = User.where(username: post_user_params[:username]).first
+          expect(new_user.staged?).to eq(true)
+        end
+
+        it "won't create the developer as staged" do
+          UsernameCheckerService.expects(:is_developer?).returns(true)
+          post "/u.json", params: post_user_params.merge(staged: true, api_key: api_key.key)
+
+          new_user = User.where(username: post_user_params[:username]).first
+          expect(new_user.staged?).to eq(false)
+        end
+      end
+    end
+
+    context 'when creating an active user (confirmed email)' do
+      before { User.any_instance.stubs(:active?).returns(true) }
+
+      it 'enqueues a welcome email' do
+        User.any_instance.expects(:enqueue_welcome_message).with('welcome_user')
+        post_user
+
+        # should save user_created_message in session
+        expect(session["user_created_message"]).to be_present
+        expect(session[SessionController::ACTIVATE_USER_KEY]).to be_present
+      end
+
+      it "shows the 'active' message" do
+        User.any_instance.expects(:enqueue_welcome_message)
+        post_user
+        expect(JSON.parse(response.body)['message']).to eq(
+          I18n.t 'login.active'
+        )
+      end
+
+      it "should be logged in" do
+        User.any_instance.expects(:enqueue_welcome_message)
+        post_user
+        expect(session[:current_user_id]).to be_present
+      end
+
+      it 'indicates the user is active in the response' do
+        User.any_instance.expects(:enqueue_welcome_message)
+        post_user
+        expect(JSON.parse(response.body)['active']).to be_truthy
+      end
+
+      it 'returns 500 status when new registrations are disabled' do
+        SiteSetting.allow_new_registrations = false
+
+        post_user
```

Comment:
It looks like we're not testing for the response status in alot of the tests. Let's sweep through once and ensure all tests assert for the response status.